cetmix_tower_server/models/cx_tower_server_template.py

```diff
@@ -299,76 +311,120 @@ def _prepare_server_values(self, **kwargs):
 
                     values[field] = new_records
 
-            # custom specific variable values
+            # Handle configuration variables if provided.
             configuration_variables = kwargs.pop("configuration_variables", None)
-            line_ids_variables = kwargs.pop("line_ids_variables", None)
+            configuration_variable_options = kwargs.pop(
+                "configuration_variable_options", {}
+            )
+
             if configuration_variables:
                 # Validate required variables
                 self._validate_required_variables(configuration_variables)
 
-                variable_vals_list = []
+                # Search for existing variable options.
+                option_references = list(configuration_variable_options.values())
+                existing_options = option_references and self.env[
+                    "cx.tower.variable.option"
+                ].search([("reference", "in", option_references)])
+                missing_options = list(
+                    set(option_references)
+                    - {option.reference for option in existing_options}
+                )
+
+                if missing_options:
+                    raise UserError(
+                        _(
+                            "Option with references '%(references)s' "
```

Comment:
Suppose we have 5 Variables with options and 3 of them are created with invalid options.  How can a user understand which option is not correct for which variable if this user doesn't have access to code?